darshan-util/pydarshan/darshan/report.py

```diff
@@ -563,14 +576,17 @@ def read_all_dxt_records(self, reads=True, writes=True, dtype=None):
             None
         """
 
+        if dtype is not None:
+            warn(f"The use of the dtype argument is deprecated in v3.3.2 and will be removed in v3.4.x", DeprecationWarning, stacklevel=2)
+
         dtype = dtype if dtype else self.dtype
 
         for mod in self.data['modules']:
-            self.mod_read_all_dxt_records(mod, warnings=False, reads=reads, writes=writes, dtype=dtype)
+            self.mod_read_all_dxt_records(mod, _warnings=False, reads=reads, writes=writes, dtype=dtype)
 
 
 
-    def mod_read_all_records(self, mod, dtype=None, warnings=True):
+    def mod_read_all_records(self, mod, dtype=None, _warnings=True):
```

Comment:
Prefixing a parameter name with a `_` is perhaps a bit unusual, though it doesn't really matter. I suppose the convention to use `_` in connection with informal privacy might confuse some.

Reply:
Hm, yeah fair point. I have revertet the change, although I think the intention of changing this parameter is actually for a user to stumble and maybe have a look at the documentation that would discourage them from populating this variable if they don't know what they are doing. We added this initially to allow the suppression of warnings by the report objects. But maybe this is in fact bad style to begin with.